test/built-ins/String/prototype/match/invoke-builtin-match.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invocation of @@match property of internally-created RegExps
+es6id: 21.1.3.11
+info: >
+    [...]
+    6. Let rx be RegExpCreate(regexp, undefined) (see 21.2.3.2.3).
+    7. ReturnIfAbrupt(rx).
+    8. Return Invoke(rx, @@match, «S»).
+features: [Symbol.match]
+---*/
+
+var originalMatch = RegExp.prototype[Symbol.match];
+var returnVal = {};
+var result, thisVal, args;
+
+RegExp.prototype[Symbol.match] = function() {
+  thisVal = this;
+  args = arguments;
+  return returnVal;
+};
+
+try {
+  result = 'target'.match('string source');
+
+  assert(thisVal instanceof RegExp);
+  assert.sameValue(thisVal.source, 'string source');
+  assert.sameValue(thisVal.flags, '');
```

Comment:
Since `lastIndex` is supposed to be set to `0` by `RegExpCreate`, could you add an assertion also for that?